example/app/index.tsx

```diff
@@ -15,45 +25,58 @@ const initialState = {
 }
 
 export default function Page() {
+  const [imageUrl, setImageUrl] = useState(yunaUrl)
   const [colors, setColors] = useState(initialState)
   const [loading, setLoading] = useState(true)
 
   useEffect(() => {
     const fetchColors = async () => {
-      const result = await getColors(yunaUrl, {
-        fallback: '#000000',
-        pixelSpacing: 5,
-      })
-
-      switch (result.platform) {
-        case 'android':
-        case 'web':
-          setColors({
-            colorOne: { value: result.lightVibrant, name: 'lightVibrant' },
-            colorTwo: { value: result.dominant, name: 'dominant' },
-            colorThree: { value: result.vibrant, name: 'vibrant' },
-            colorFour: { value: result.darkVibrant, name: 'darkVibrant' },
-            rawResult: JSON.stringify(result),
-          })
-          break
-        case 'ios':
-          setColors({
-            colorOne: { value: result.background, name: 'background' },
-            colorTwo: { value: result.detail, name: 'detail' },
-            colorThree: { value: result.primary, name: 'primary' },
-            colorFour: { value: result.secondary, name: 'secondary' },
-            rawResult: JSON.stringify(result),
-          })
-          break
-        default:
-          throw new Error('Unexpected platform')
+      try {
+        const result = await getColors(imageUrl, {
+          fallback: '#000000',
+          pixelSpacing: 5,
+        })
+        switch (result.platform) {
+          case 'android':
+          case 'web':
+            setColors({
+              colorOne: { value: result.lightVibrant, name: 'lightVibrant' },
+              colorTwo: { value: result.dominant, name: 'dominant' },
+              colorThree: { value: result.vibrant, name: 'vibrant' },
+              colorFour: { value: result.darkVibrant, name: 'darkVibrant' },
+              rawResult: JSON.stringify(result),
+            })
+            break
+          case 'ios':
+            setColors({
+              colorOne: { value: result.background, name: 'background' },
+              colorTwo: { value: result.detail, name: 'detail' },
+              colorThree: { value: result.primary, name: 'primary' },
+              colorFour: { value: result.secondary, name: 'secondary' },
+              rawResult: JSON.stringify(result),
+            })
+            break
+          default:
+            throw new Error('Unexpected platform')
+        }
+      } catch (error) {
+        console.error(error)
       }
 
       setLoading(false)
     }
 
     fetchColors()
-  }, [])
+  }, [imageUrl])
+
+  const getContentFile = async () => {
+    const res = await DocumentPicker.getDocumentAsync({
+      type: 'image/*',
+      copyToCacheDirectory: false, // 关键：不要拷贝，否则就是 file://
+    })
+    if (res.canceled) return
+    setImageUrl(res.assets?.[0].uri || '')
```

Comment:
awesome! 
one point of feedback here: It's not a good practice to observe React state in a `useEffect`. Could we refactor `fetchColors` to take in the image URL variable instead?